compliance/unapproved_instance_types/unapproved_instance_types.pt

```diff
@@ -0,0 +1,191 @@
+name "Unapproved instance types"
+rs_pt_ver 20180301
+type "policy"
+short_description "Report on any instances that are running using instance types that are not approved. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/compliance/approved_instance_types) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
+long_description "Version: 1.0"
+category "Compliance"
+severity "medium"
+
+###############################################################################
+# User inputs
+###############################################################################
+
+parameter "param_email" do
+  type "list"
+  label "Email addresses to notify"
+  description "Email addresses of the recipients you wish to notify when new incidents are created"
+end
+
+parameter "param_approved_instance_types" do
+  type "list"
+  label "Approved Instance Types"
+  description "List of approved instance types of AWS, Azure and Google cloud, separated by comma. Example: 'a1.medium,a1.large,a1.xlarge,Standard_A0,Standard_A6,f1-micro,n1-highcpu-2 etc..'"
```

Comment:
The list types doesn't need or use commas.  I don't think the commas work, but you could correct me if I'm wrong.

Reply:
@cdwilhelm. True, the list type doesn't require a comma separated. But I thought when there is more number of approved instance types need to enter as an input, the user may find it difficult to enter all instance types one-by-one(without comma separated). In this case, the user can construct an instance types list by a comma-separated and enter as a single parameter. 
To handle comma separated input. In policy, I am checking the instance's instance type with both list or comma separated string of input parameter. I tested with both cases and its working as expected.
if((param_approved_instance_types.indexOf(instance_type) !== -1) || (param_approved_instance_types.join()).match(instance_type)){
        is_instance_type_matched = true;
 }
I can change it if this is not the proper way of handling the input parameter.

Reply:
Running template link for instances input parameter as list:
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5d076945dcd1a5008130b5ea

Running template link for instance types input parameter as comma separated string:
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5d07650f64edc40081fa75aa